src/responder/kcm/kcm.c

```diff
@@ -347,14 +395,49 @@ int main(int argc, const char *argv[])
 
     poptFreeContext(pc);
 
+    tmp_ctx = talloc_new(NULL);
+    if (!tmp_ctx) {
+        return 3;
+    }
+
     /* set up things like debug, signals, daemonization, etc. */
     debug_log_file = "sssd_kcm";
     DEBUG_INIT(debug_level, opt_logger);
 
-    ret = server_setup("kcm", true, 0, uid, gid, CONFDB_FILE,
+     if (opt_config_file == NULL) {
+        config_file = SSSD_CONFIG_FILE;
+    } else {
+        config_file = opt_config_file;
+    }
+
+   /* Parse config file, fail if cannot be done */
+    ret = load_configuration(config_file, CONFDB_DEFAULT_CONFIG_DIR, "kcm");
+    if (ret != EOK) {
+        switch (ret) {
+        case EPERM:
+        case EACCES:
```

Comment:
I see no point in this `switch`.
It's enough to log `sss_strerror(ret)`
Removing `switch` will allow to avoid about proper wording in `CONF_FILE_PERM_ERROR_MSG` (I mean 'owned by root.root', this is not the case already).

Reply:
`switch` remove.

Reply:
`CONF_FILE_PERM_ERROR_MSG` should be removed as well in this file.

Reply:
Already done, but waiting for a complete set of changes before pushing.